tests/test_end_to_end.py

```diff
@@ -172,3 +172,32 @@ def test_reference_grib(
     # various inconsistencies (of dtype casting int to float, etc.). With the right combination of
     # options passed to the pipeline, seems like these should pass?
     # xr.testing.assert_equal(ds.load(), ds2)
+
+
+@pytest.mark.parametrize("consolidate_dimension_coordinates", [False, True])
+def test_xarray_zarr_consolidate_dimension_coordinates(
+    netcdf_local_file_pattern_sequential,
+    pipeline,
+    tmp_target,
+    consolidate_dimension_coordinates,
+):
+    pattern = netcdf_local_file_pattern_sequential
+    with pipeline as p:
+        (
+            p
+            | beam.Create(pattern.items())
+            | OpenWithXarray(file_type=pattern.file_type)
+            | StoreToZarr(
+                target_root=tmp_target,
+                store_name="subpath",
+                combine_dims=pattern.combine_dim_keys,
+                consolidate_dimension_coordinates=consolidate_dimension_coordinates,
+            )
+        )
+
```

Comment:
```suggestion
        unconsolidated_zarr = (
            p
            | beam.Create(pattern.items())
            | OpenWithXarray(file_type=pattern.file_type)
            | StoreToZarr(
                target_root=tmp_target,
                store_name="subpath",
                combine_dims=pattern.combine_dim_keys,
            )
        )
        if consolidate_dimension_coordinates:
            unconsolidated_zarr | ConsolidateCoordinateDimensions()

```

Assuming we go with the composable/modular design rather than the boolean flag design, then this test would need to updated to look something like above.

I'm not sure if `ConsolidateMetadata()` would also need to be piped on the end here to get the asserts below to pass.